src/main/resources/com/ncc/aif/restricted_hypothesis_aif.shacl

```diff
@@ -123,6 +123,33 @@ aida:HypothesisEntityClusterRequireHandle
         sh:minCount 1 ;
     ] .
 
+########################
+# 4.3 TA3 #7  Clusters must be homogeneous by base class (Entity, Event, or Relation)
+#------------------------
+aida:HypothesisClusterMembersSameAsBaseClass
+    a sh:SPARQLConstraint ;
+    sh:message "All Clusters must be homogeneous by base class (Entity, Event, or Relation). Found {?value}" ;
+    sh:select """
+        PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+        PREFIX aida:  <https://tac.nist.gov/tracks/SM-KBP/2018/ontologies/InterchangeOntology#>
+        SELECT $this (COUNT(?cluster) AS ?value)
```

Comment:
Can't tell from the return value which cluster or Entity/Event/Relation violates this. Rather than a count, this should probably indicate which Entity/Event/Relation doesn't match which cluster.

Reply:
Updated.